app/controllers/users_controller.rb

```diff
@@ -0,0 +1,12 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all.order(:created_at)
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    if @user == nil
+      redirect_to "/404.html"
```

Comment:
instead of hardcoding this `redirect to "/404.html"` I would rather you raise a built in Rails error to render the 404 page